src/orquestra/integrations/qiskit/backend/backend.py

```diff
@@ -320,17 +323,15 @@ def aggregate_measurements(
                 circuit_index += 1
 
             if self.readout_correction:
-                current_circuit = circuit_set[circuit_index - 1]
-                active_qubits = list(
-                    {
-                        qubit.index
-                        for inst in current_circuit.data
-                        if isinstance(inst[0], QiskitGate)
-                        for qubit in inst[1]
-                    }
+                current_circuit_from_jobs = circuit_set_from_jobs[circuit_index - 1]
+                current_circuit_from_batches = circuit_set_from_batches[
+                    circuit_index - 1
+                ]
+                virtual_to_physical_qubits_dict = _get_virtual_to_physical_qubits_dict(
+                    current_circuit_from_batches, current_circuit_from_jobs
                 )
                 combined_counts = self._apply_readout_correction(
```

Comment:
Suppose I have an experiment of 50 circuits with 1000 shots per circuit. The number of circuits and shots per circuit are less than IBM backend max limits. 

Will the `_apply_readout_correction` method be called for each circuit? If not, ignore the remaining comment.

If yes, that might be optimized as we can run the measurement calibration once and reuse that calibration data for each of these 50 circuits.

One catch is that `physical_qubits` list may be different for different circuits. In that case, we can run measurement calibration once for the full `QuantumRegister` (qr) and then, use Qiskit's `subset_fitter`. Again, the decision will depend on 1 full `qr` calibration vs. for this example, 50 partial calibrations.

The ultimate goal will be optimizing (minimizing) IBM backend usage. 

Example: 
Suppose, we have a 7 qubit device and the experiments contains 50 circuits each consisting 3 qubits. 
1 full 7 qubit calibration needs 2^7 = 128 calibration circuits.
50 circuits each with 3 qubits need 50 x 2^3 = 50 x 8 = 400 circuits. 

For this example, running measurement calibration once for full device and using `subset_fiiter` is more economical.

Reply:
`_apply_readout_correction ` is called for each circuit, but the calibration is only run once for each set of qubits because `readout_correction_filters`  saves the filters.